ur_calibration/README.md

```diff
@@ -0,0 +1,77 @@
+# ur_calibration
+
+Package for extracting the factory calibration from a UR robot and changing it to be used by `ur_description` to gain a correct URDF model.
+
+Each UR robot is calibrated inside the factory giving exact forward and inverse kinematics. To also
+make use of this in ROS, you first have to extract the calibration information from the robot.
+
+Though this step is not necessary, to control the robot using this driver, it is highly recommended
+to do so, as end effector positions might be off in the magnitude of centimeters.
+
+## Nodes
+### calibration_correction
+This node extracts calibration information directly from a robot, calculates the URDF correction and
+saves it into a .yaml file.
+
+In the launch folder of the ur_calibration package is a helper script:
+
+```bash
+$ ros2 launch ur_calibration calibration_correction.py.launch \
+robot_ip:=<robot_ip> target_filename:="${HOME}/my_robot_calibration.yaml"
+```
+
+For the parameter `robot_ip` insert the IP address on which the ROS pc can reach the robot. As
+`target_filename` provide an absolute path where the result will be saved to.
+
+## Creating a calibration / launch package for all local robots
+When dealing with multiple robots in one organization it might make sense to store calibration data
+into a package dedicated to that purpose only. To do so, create a new package (if it doesn't already
+exist)
+
+```bash
+# Replace your actual colcon_ws folder
+$ cd <colcon_ws>/src
+$ ros2 pkg  create example_organization_ur_launch --build-type ament_cmake  --dependencies ur_client_library \
+--description "Package containing calibrations and launch files for our UR robots."
+# Create a skeleton package
+$ mkdir -p example_organization_ur_launch/etc
+$ mkdir -p example_organization_ur_launch/launch
+$ echo 'install(DIRECTORY etc launch DESTINATION share/${PROJECT_NAME})' >> example_organization_ur_launch/CMakeLists.txt
+$ colcon build --packages-select example_organization_ur_launch
+```
+
+We can use the new package to store the calibration data in that package. We recommend naming each
+robot individually, e.g. *ex-ur10-1*.
+
+```bash
+$ ros2 launch ur_calibration calibration_correction.py.launch \
+robot_ip:=<robot_ip> \
+target_filename:="$(ros2 pkg prefix example_organization_ur_launch)/share/example_organization_ur_launch/etc/ex-ur10-1_calibration.yaml"
+```
+
+To make life easier, we create a launchfile for this particular robot. We base it upon the
+respective launchfile in the driver:
+
+```bash
+# Replace your actual colcon_ws folder
+$ cd <colcon_ws>/src/example_organization_ur_launch/launch
+$ cp $(ros2 pkg prefix ur_bringup)/share/ur_bringup/launch/ur_control.launch.py ex-ur10-1.launch.py
+```
+
+Next, modify the parameter section of the new launchfile to match your actual calibration:
+
+```py
+kinematics_params = PathJoinSubstitution(
+        [FindPackageShare("example_organization_ur_launch"), "etc", "", "ex-ur10-1_calibration.yaml"]
```

Comment:
```suggestion
        [FindPackageShare("<organization_name>_ur_launch"), "etc", "", "ex-ur10-1_calibration.yaml"]
```

Reply:
Should we enable to set "calibration_package" as argument to launch files?

Reply:
I would leave it like this if we are going to follow the advice [here](https://github.com/UniversalRobots/Universal_Robots_ROS2_Driver/pull/186/files#r718573729)